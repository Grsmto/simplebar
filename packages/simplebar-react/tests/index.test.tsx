import * as React from 'react';
import { render, waitFor } from '@testing-library/react';
import SimpleBar from '../index';
import SimpleBarCore from 'simplebar-core';

test('renders without crashing', () => {
  const { container } = render(
    <div>
      <SimpleBar>
        {[...Array(5)].map((_, i) => (
          <p key={i}>Some content</p>
        ))}
      </SimpleBar>
    </div>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('renders with options', () => {
  const { container } = render(
    <SimpleBar forceVisible="y" tabIndex={-1}>
      {[...Array(5)].map((x, i) => (
        <p key={i}>Some content</p>
      ))}
    </SimpleBar>
  );
  expect(container.firstChild).toMatchSnapshot();

  expect(
    container.querySelector('.simplebar-track.simplebar-vertical')
  ).toBeVisible();
});

test('renders with object as option', () => {
  const { container } = render(
    <SimpleBar
      classNames={{
        wrapper: 'custom-class',
      }}
    >
      {[...Array(5)].map((x, i) => (
        <p key={i}>Some content</p>
      ))}
    </SimpleBar>
  );
  expect(container.firstChild).toMatchSnapshot();

  expect(container.querySelector('.custom-class')).toBeVisible();
});

test('renders with scrollableNodeProps', async () => {
  const { getByTestId } = render(
    <SimpleBar
      scrollableNodeProps={{
        'data-testid': 'scrollable-node-props',
      }}
    >
      {[...Array(5)].map((x, i) => (
        <p key={i}>Some content</p>
      ))}
    </SimpleBar>
  );

  await waitFor(() =>
    // getByTestId throws an error if it cannot find an element
    getByTestId('scrollable-node-props')
  );
});

test('renders with ref', async () => {
  const ref = React.createRef<SimpleBarCore | null>();

  const { unmount } = render(
    <SimpleBar
      ref={(instance) => {
        //
      }}
    >
      {[...Array(5)].map((x, i) => (
        <p key={i}>Some content</p>
      ))}
    </SimpleBar>
  );

  if (!ref.current) return;

  jest.spyOn(ref.current, 'unMount');

  unmount();

  expect(ref.current.unMount).toHaveBeenCalled();
});

test('works on unmount', async () => {
  const ref = React.createRef<any>();

  const { unmount } = render(
    <SimpleBar ref={ref} style={{ maxHeight: 300 }}>
      {[...Array(5)].map((x, i) => (
        <p key={i}>Some content</p>
      ))}
    </SimpleBar>
  );

  jest.spyOn(ref.current, 'unMount');

  unmount();

  expect(ref.current.unMount).toHaveBeenCalled();
});

test('renders with ref callback', async () => {
  const callback = jest.fn((node) => {
    return node;
  });

  const { unmount } = render(
    <SimpleBar ref={callback}>
      {[...Array(5)].map((x, i) => (
        <p key={i}>Some content</p>
      ))}
    </SimpleBar>
  );

  expect(callback).toHaveBeenCalled();
  unmount();
  expect(callback).toHaveBeenCalledWith(null);
});

test('renders with render function', async () => {
  let ref1;
  let ref2;

  render(
    <SimpleBar>
      {({ scrollableNodeRef, contentNodeRef }) => {
        ref1 = scrollableNodeRef;
        ref2 = contentNodeRef;
        return null;
      }}
    </SimpleBar>
  );

  expect(ref1).toBeDefined();
  expect(ref2).toBeDefined();
});
