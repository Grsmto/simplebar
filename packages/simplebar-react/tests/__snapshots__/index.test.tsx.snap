// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with object as option 1`] = `
<div
  data-simplebar="init"
>
  <div
    class="custom-class"
  >
    <div
      class="simplebar-height-auto-observer-wrapper"
    >
      <div
        class="simplebar-height-auto-observer"
      />
    </div>
    <div
      class="simplebar-mask"
    >
      <div
        class="simplebar-offset"
        style="right: 0px; bottom: 0px;"
      >
        <div
          aria-label="scrollable content"
          class="simplebar-content-wrapper"
          role="region"
          style="height: auto; overflow-x: hidden; overflow-y: hidden;"
          tabindex="0"
        >
          <div
            class="simplebar-content"
          >
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="simplebar-placeholder"
      style="width: 0px; height: 0px;"
    />
  </div>
  <div
    class="simplebar-track simplebar-horizontal"
    style="visibility: hidden;"
  >
    <div
      class="simplebar-scrollbar"
      style="width: 0px; display: none;"
    />
  </div>
  <div
    class="simplebar-track simplebar-vertical"
    style="visibility: hidden;"
  >
    <div
      class="simplebar-scrollbar"
      style="height: 0px; display: none;"
    />
  </div>
</div>
`;

exports[`renders with options 1`] = `
<div
  class="simplebar-scrollable-y"
  data-simplebar="init"
>
  <div
    class="simplebar-wrapper"
  >
    <div
      class="simplebar-height-auto-observer-wrapper"
    >
      <div
        class="simplebar-height-auto-observer"
      />
    </div>
    <div
      class="simplebar-mask"
    >
      <div
        class="simplebar-offset"
        style="right: -0px; bottom: 0px;"
      >
        <div
          aria-label="scrollable content"
          class="simplebar-content-wrapper"
          role="region"
          style="height: auto; overflow-x: hidden; overflow-y: scroll;"
          tabindex="-1"
        >
          <div
            class="simplebar-content"
          >
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="simplebar-placeholder"
      style="width: 0px; height: 0px;"
    />
  </div>
  <div
    class="simplebar-track simplebar-horizontal"
    style="visibility: hidden;"
  >
    <div
      class="simplebar-scrollbar"
      style="width: 0px; display: none;"
    />
  </div>
  <div
    class="simplebar-track simplebar-vertical"
    style="visibility: visible;"
  >
    <div
      class="simplebar-scrollbar"
      style="height: 0px; display: none;"
    />
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  <div
    data-simplebar="init"
  >
    <div
      class="simplebar-wrapper"
    >
      <div
        class="simplebar-height-auto-observer-wrapper"
      >
        <div
          class="simplebar-height-auto-observer"
        />
      </div>
      <div
        class="simplebar-mask"
      >
        <div
          class="simplebar-offset"
          style="right: 0px; bottom: 0px;"
        >
          <div
            aria-label="scrollable content"
            class="simplebar-content-wrapper"
            role="region"
            style="height: auto; overflow-x: hidden; overflow-y: hidden;"
            tabindex="0"
          >
            <div
              class="simplebar-content"
            >
              <p>
                Some content
              </p>
              <p>
                Some content
              </p>
              <p>
                Some content
              </p>
              <p>
                Some content
              </p>
              <p>
                Some content
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="simplebar-placeholder"
        style="width: 0px; height: 0px;"
      />
    </div>
    <div
      class="simplebar-track simplebar-horizontal"
      style="visibility: hidden;"
    >
      <div
        class="simplebar-scrollbar"
        style="width: 0px; display: none;"
      />
    </div>
    <div
      class="simplebar-track simplebar-vertical"
      style="visibility: hidden;"
    >
      <div
        class="simplebar-scrollbar"
        style="height: 0px; display: none;"
      />
    </div>
  </div>
</div>
`;
